Content.Client/GameObjects/Components/Doors/ClientDoorComponent.cs

```diff
@@ -0,0 +1,74 @@
+#nullable enable
+using Content.Shared.GameObjects.Components.Doors;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Robust.Shared.ViewVariables;
+using System;
+
+namespace Content.Client.GameObjects.Components.Doors
+{
+    /// <summary>
+    /// Bare-bones client-side door component; used to stop door-based mispredicts.
+    /// </summary>
+    [UsedImplicitly]
+    [RegisterComponent]
+    public class ClientDoorComponent : SharedDoorComponent
+    {
+        private bool _stateChangeHasProgressed = false;
+        private TimeSpan _timeOffset;
+
+        public override void HandleComponentState(ComponentState? curState, ComponentState? nextState)
+        {
+            base.HandleComponentState(curState, nextState);
+
+            if (curState is not DoorComponentState doorCompState)
+            {
+                return;
+            }
+
+            CurrentlyCrushing = doorCompState.CurrentlyCrushing;
+            StateChangeStartTime = doorCompState.StartTime;
+            State = doorCompState.DoorState;
+
+            if (StateChangeStartTime == null)
+            {
+                return;
+            }
+
+            _timeOffset = State switch
+            {
+                DoorState.Opening => OpenTimeOne,
+                DoorState.Closing => CloseTimeOne,
+                _ => throw new ArgumentOutOfRangeException(),
+            };
+
+            if (doorCompState.CurTime >= StateChangeStartTime + _timeOffset)
```

Comment:
Pretty sure you don't need to send CurTime if the client already knows when a transition is happening like Acruid said...

Reply:
Well, the first time the `DoorComponentState` gets applied, its `CurTime` is in sync with the client's. Later, though, the client's `CurTime` has advanced while it's still applying the same `DoorComponentState`... I didn't see this happen without the state being applied at the correct `CurTime` first, though, so as long as the `StateChangeStartTime` is changing with the new state, I can use `GameTiming.CurTime`... my initial fear was that `CurTime` wouldn't be in sync with the state's, but luckily I can work around it here.